### PR TITLE
V0.1

### LINKED ISSUE #10: Write tests for utils in v0.1
Write a test checking the implementation of starkent_keccak. 

You can use this test : 

```
        >>> from nethermind.starknet_abi.utils import starknet_keccak
        >>> starknet_keccak(b"transfer").hex()
        '0083afd3f4caedc6eebf44246fe54e38c95e3179a5ec9ea81740eca5b482d12e'

```

### PR DESCRIPTION
**Pull request by Rishikpulhani**
fixed #10

### CODE REVIEW

File: athena_abi/utils_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+func TestStarknetKeccak(t *testing.T) {
+	// Known input and output based on Keccak-256 hashing and masking
+	tests := []struct {
+		input    []byte
+		expected []byte
+	}{
+		{[]byte("transfer"), []byte{0x00, 0x83, 0xaf, 0xd3, 0xf4, 0xca, 0xed, 0xc6, 0xee, 0xbf, 0x44, 0x24, 0x6f, 0xe5, 0x4e, 0x38, 0xc9, 0x5e, 0x31, 0x79, 0xa5, 0xec, 0x9e, 0xa8, 0x17, 0x40, 0xec, 0xa5, 0xb4, 0x82, 0xd1, 0x2e}}, // Expected output from Keccak-256 hashing of "transfer"
+	}
+
+	for _, tt := range tests {
```
Comment (Sh0g0-1758) [inline athena_abi/utils_test.go]: There is only one test no ?

File: athena_abi/utils_test.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	t.Run("Empty Input", func(t *testing.T) {
+		result := StarknetKeccak([]byte{})
+		expected := []byte{0x1, 0xd2, 0x46, 0x1, 0x86, 0xf7, 0x23, 0x3c, 0x92, 0x7e, 0x7d, 0xb2, 0xdc, 0xc7, 0x3, 0xc0, 0xe5, 0x0, 0xb6, 0x53, 0xca, 0x82, 0x27, 0x3b, 0x7b, 0xfa, 0xd8, 0x4, 0x5d, 0x85, 0xa4, 0x70} // Calculate this in advance if known or check with hashing tool
+		assert.Equal(t, expected, result)
+	})
+
+	t.Run("Large Input", func(t *testing.T) {
+		largeInput := make([]byte, 1024) // 1 KB of data
+		result := StarknetKeccak(largeInput)
+		expected := StarknetKeccak(largeInput) // Calculate this in advance if known or check with hashing tool
```
Comment (Sh0g0-1758) [inline athena_abi/utils_test.go]: are you using the same function for both result and expected ?

File: athena_abi/utils_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+	t.Run("Exact Length Match", func(t *testing.T) {
+		result := bigIntToBytes(*big.NewInt(255), 1)
+		assert.Equal(t, []byte{0xFF}, result)
+	})
+
+	t.Run("Large Integer", func(t *testing.T) {
+		largeInt := big.NewInt(0).Lsh(big.NewInt(1), 64) // 2^64
+		result := bigIntToBytes(*largeInt, 9)
+		expected := []byte{0x01, 0, 0, 0, 0, 0, 0, 0, 0} // 2^64 in 8 bytes
```
Comment (Sh0g0-1758) [inline athena_abi/utils_test.go]: Are these not 9 bytes ?

**Review by Sh0g0-1758 — CHANGES_REQUESTED** [review]

File: athena_abi/utils_test.go (hunk 2 above)
Comment (Rishikpulhani) [reply athena_abi/utils_test.go]: Sir sorry actually I forgot to remove that test

File: athena_abi/utils_test.go (hunk 1 above)
Comment (Rishikpulhani) [reply athena_abi/utils_test.go]: Sir I'll add some more

File: athena_abi/utils_test.go (hunk 3 above)
Comment (Rishikpulhani) [reply athena_abi/utils_test.go]: Sir I didn't change the comments 
I'll do make the changes

File: athena_abi/utils_test.go (hunk 1 above)
Comment (Rishikpulhani) [reply athena_abi/utils_test.go]: sir there are some more test cases below
should i add some more

Comment (Sh0g0-1758) [reply athena_abi/utils_test.go]: well no tests is an array of only one test no ? So this for loop is redundant ?

File: athena_abi/utils_test.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+		t.Run("Normal Case", func(t *testing.T) {
+			result := StarknetKeccak(tt.input)
+			assert.Equal(t, tt.expected, result)
+		})
+	}
+
+	// Boundary cases
+	t.Run("Empty Input", func(t *testing.T) {
+		result := StarknetKeccak([]byte{})
+		expected := []byte{0x1, 0xd2, 0x46, 0x1, 0x86, 0xf7, 0x23, 0x3c, 0x92, 0x7e, 0x7d, 0xb2, 0xdc, 0xc7, 0x3, 0xc0, 0xe5, 0x0, 0xb6, 0x53, 0xca, 0x82, 0x27, 0x3b, 0x7b, 0xfa, 0xd8, 0x4, 0x5d, 0x85, 0xa4, 0x70} // Calculate this in advance if known or check with hashing tool
```
Comment (Sh0g0-1758) [inline athena_abi/utils_test.go]: Can you remove this comment above and below.

File: athena_abi/utils_test.go (hunk 1 above)
Comment (Rishikpulhani) [reply athena_abi/utils_test.go]: ohh ok sir ill change it

**Review by Sh0g0-1758 — APPROVED** [review]
LGTM. Looks good and exhaustive. Thanks for the contribution.

**Pull request merged**